### ISSUE TITLE
Error while embedding stories

### ISSUE THREAD

**Issue by chaithalbe** [JeremyRH/storybook-addon-code-editor#44]
Hi, I am getting this error while [embedding stories](https://storybook.js.org/docs/sharing/embed)(I am not using chromatic) with this add-on. 

When I revert the stories and write them like mentioned in the [Storybook docs](https://storybook.js.org/docs/writing-stories) it works fine.

```
manager-bundle.js:18 [Storybook] One of your manager-entries failed: http://localhost:6006/sb-addons/code-editor-1/manager-bundle.js DOMException: Failed to read a named property '_addon_code_editor_store' from 'Window': Blocked a frame with origin "http://localhost:6006" from accessing a cross-origin frame.
    at z (http://localhost:6006/sb-addons/code-editor-1/manager-bundle.js:17:2267)
    at http://localhost:6006/sb-addons/code-editor-1/manager-bundle.js:17:3011
    at http://localhost:6006/sb-addons/code-editor-1/manager-bundle.js:17:4920
(anonymous) @ manager-bundle.js:18

runtime.js:4 DOMException: Failed to read a named property '_addon_code_editor_store' from 'Window': Blocked a frame with origin "http://localhost:6006" from accessing a cross-origin frame.
    at createStore (http://localhost:6006/node_modules/.cache/storybook/1c3385a5d25e538d10b518b310c74d3ca2690b6aaffeadccd74da79736171f86/sb-vite/deps/storybook-addon-code-editor.js?v=7009db18:125570:28)
    at http://localhost:6006/node_modules/.cache/storybook/1c3385a5d25e538d10b518b310c74d3ca2690b6aaffeadccd74da79736171f86/sb-vite/deps/storybook-addon-code-editor.js?v=7009db18:125621:13
```

Versions:
storybook-addon-code-editor: 3.0.1
storybook: 8.0.10

**Comment by JeremyRH** [comment]
Thanks for creating the issue.

Storybook uses iframes for displaying the preview of a story. This allows content to be "sandboxed" and not affected by other parts of Storybook. In order to share state between the content iframe and this addon's frame, [I used `window.top`](https://github.com/JeremyRH/storybook-addon-code-editor/blob/1166b222a0f6b2b7cc693b56930b5046f158637a/src/createStore.ts#L51).

This should probably be changed to use Storybooks' [Channels](https://storybook.js.org/docs/addons/configure-addons#channels). Last time I tried to use channels, events could be missed depending on a few things so I went with using `window.top`. I could also try using `window.parent` but I have a feeling there was a reason I used `window.top`.

**Comment by chaithalbe** [comment]
Ok, will there be a fix for this? If not, is there a workaround?

**Comment by chaithalbe** [comment]
Hey @JeremyRH , just checking in to see if you have any updates on this issue or an estimated timeline for this fix?

**Comment by JeremyRH** [comment]
I'll publish a version using `window.parent` if `window.top` fails. That should be safe and fix the issue with embedded stories.

**Comment by JeremyRH** [comment]
I made a mistake. The preview and manager are both using the same code. So accessing `window.parent` raises an exception in the manager. This is why I was using `window.top`, I assumed it would be the same window context no matter if the preview or manager used that. I'll have another fix soon.
Sorry and thanks again for the help testing this.

**Comment by JeremyRH** [comment]
Tested the fix in 3.0.3. Everything seems to work!
